src/Reference/ContainsOne.php

```diff
@@ -60,14 +57,18 @@ protected function init(): void
         }
 
         $ourModel = $this->getOurModel();
+        $ourField = $this->getOurFieldName();
 
-        if (!$ourModel->hasElement($this->our_field)) {
-            $ourModel->addField($this->our_field, [
+        if (!$ourModel->hasElement($ourField)) {
+            $ourModel->addField($ourField, [
                 'type' => $this->type,
                 'reference' => $this,
                 'system' => $this->system,
                 'caption' => $this->caption, // it's ref models caption, but we can use it here for field too
-                'ui' => $this->ui,
+                'ui' => array_merge_recursive([
```

Comment:
array_merge_recursive DOES NOT have any affect here -> remove

Reply:
`ui` is multi-level array property. array_merge will merge only 1st level of array and overwrite all other levels, so it's not possible to extend `ui` with `ui => ['form' => [AnotherClass::class]]` for example without breaking other (maybe default ones) ui properties.

OH crap, you're right. It should be `array_replace_recursive` here - I will fix.

Reply:
I did not read, but here the 1st arg is NOT nested arrray, thus I do not understand `_recursive`, seems like a typo to me